eng/pipelines/common/macos-sign-with-entitlements.yml

```diff
@@ -0,0 +1,52 @@
+parameters:
+  fileToSign : ''
```

Comment:
Can this take a list:
```yml
parameters
  filesToSign:[]
```

and then the caller can do:
```yml
- template: /eng/pipelines/common/macos-sign-with-entitlements.yml
  parameters:
    filesToSign:
      - fileToSign: createdump
        fileToSignFolder: $(buildProductRootFolderPath)
        entitlementsFile: $(Build.SourcesDirectory)/eng/pipelines/common/createdump-entitlements.plist
      - fileToSign: corerun
        fileToSignFolder: $(buildProductRootFolderPath)
        entitlementsFile: $(Build.SourcesDirectory)/eng/pipelines/common/entitlements.plist
```
And notarize them and then package them all in one zip and just send one ESRP request?

Reply:
How does the template use "filesToSign"?  Like the `codesign ... ` command that can only take one file path at a time? Is it like msbuild metadata on a property?

Reply:
Call site:
```yml
- template: /eng/pipelines/common/macos-sign-with-entitlements.yml
  parameters:
    filesToSign:
      - name: createdump
        directory: $(buildProductRootFolderPath)
        entitlementsFile: $(Build.SourcesDirectory)/eng/pipelines/common/createdump-entitlements.plist
      - name: corerun
        directory: $(buildProductRootFolderPath)
        entitlementsFile: $(Build.SourcesDirectory)/eng/pipelines/common/entitlements.plist
```
the mac-os-file:
```yml
parameters
  filesToSign:[]

- ${{ each file in parameters.filesToSign }}:
  - script: codesign -s - -f --entitlements ${{ file.entitlementsFile }} ${{ file.path }}/${{ file.name }}
    displayName: 'Add entitlements to ${{ file.name }}'
```